### PR TITLE
samples: drivers: adc: Allow coverage calculation.

### PR DESCRIPTION
**Pull request by nordic-babu**
Sample must end to dump coverage data.

### CODE REVIEW

**Review by nordic-piks — APPROVED** [review]

**Review by nordic-piks — APPROVED** [review]

File: samples/drivers/adc/adc_dt/src/main.c [hunk 1]
```diff
@@ -54,7 +54,11 @@ int main(void)
 		}
 	}
 
+#ifndef CONFIG_COVERAGE
 	while (1) {
+#else
+	for (int k = 0; k < 10; k++) {
```
Comment (decsny) [inline samples/drivers/adc/adc_dt/src/main.c:60]: 10 is a magic number

**Review by decsny — CHANGES_REQUESTED** [review]
Can you help me understand what this change has to do with coverage

**Comment by nordic-piks** [comment]
> Can you help me understand what this change has to do with coverage

Sure ;)
Coverage data are dumped on console after main ends.
Sample must end to produce them so infinite loop is changed into finite one in coverage mode.

File: samples/drivers/adc/adc_dt/src/main.c (hunk 1 above)
Comment (nordic-piks) [reply samples/drivers/adc/adc_dt/src/main.c:60]: Indeed, but very similar to other changed eg. #71207
That number has no meaning, there are just needed a few loops.

Comment (nordic-babu) [reply samples/drivers/adc/adc_dt/src/main.c:60]: exactly as @nordic-piks explained, no of loops has to be finite and low enough to not exceed test timeout (together with time needed for dumping coverage data). 
Eventually we can even drop #else section and have one execution - that's enough, but I think it's nothing wrong to have few readings in output.

**Review by anangl — APPROVED** [review]

**Review by decsny — APPROVED** [review]

**Comment by decsny** [comment]
kind of ugly how this is being done but alright I am not going to fight about it right now

**Pull request merged**